### ISSUE TITLE
when the number of struct members is less than the number of columns in the table

### ISSUE THREAD

**Issue by Archercober** [taocpp/taopq#74]
For example

```
// an aggregate
struct user
{
   std::string name;
};

template<>
inline constexpr bool tao::pq::is_aggregate< user > = true;

int main()
{
   // open a connection to the database
   const auto conn = tao::pq::connection::create( "dbname=template1" );

   // execute statements
   conn->execute( "DROP TABLE IF EXISTS users" );
   conn->execute( "CREATE TABLE users ( name TEXT PRIMARY KEY, age INTEGER NOT NULL, planet TEXT NOT NULL )" );

   // execute previously prepared statements
   conn->execute( "insert_user", user{ "R. Daneel Olivaw", 19230, "Earth" } );

   // query and convert data even user only has on field
   for( const user u : conn->execute( "SELECT * FROM users" ) ) {
      std::cout << u.name;
   }
}
```
Is it possible to convert to a struct with fewer members than the number of table columns?

**Comment by d-frey** [comment]
This seems to be a bug, it works when you add a second member to `user`. The bug seems to occur when there is only a single member to bind to. I'll investigate when I have some time...

**Comment by d-frey** [comment]
Note that I consider it to be really bad practice to use `SELECT * FROM ...` as this is just begging for code to break in the future when the DB schema is modified. Always list the columns (and their order) explicitly in your SELECT statements. Anyway, that is unrelated to the bug.

**Comment by d-frey** [comment]
I fixed the single member aggregate case, but I still wonder if the code should check (obviously at runtime) for excess result columns as explained in my previous comment...

**Comment by d-frey** [comment]
Found out what the problem with my original test was, I mixed up table names and never actually tested with data, hence the check in the code was never triggered. If there is data, the check was always there, so it would always catch the excess columns at runtime. The only bug was the compile-time problem for single member aggregates you found and that is fixed now. Case closed, thank you for the report!
